### DISCUSSION TITLE
Threads

### DISCUSSION THREAD

**Discussion by ghost** [stoatchat/discussions#221] [Feature Suggestions]
Threads help organize conversations in channels better, and role icons are fancy :O

Maintainer edit: for role icons, see https://github.com/orgs/revoltchat/discussions/688

**Comment by Pukimaa** [comment]
I really do like Discord's Role Icons and the little icon that you see in the chat. Would love to see it in Revolt!

**Reply by TheActualConner** [reply]
I completely agree with this, role icons are a great feature and are especially great for identifying roles without opening profiles. Especially because on Revolt, opening someone's profile you can not see their roles.

> The Conner
> @ TheConner # 0001

**Comment by nulldg** [comment]
two feature suggestions should be two conversations.

that aside, discord implemented threads in an absolutely abysmal way, so i find people rarely using them as intended. personally i would rather not see threads in revolt at all, but if revolt does end up adding them, revolt will need to take care to not repeat the same mistakes discord did. for instance, forums put a huge dent the organization problem for large servers, where threads turned out to be disappointingly ineffective.

**Reply by Rexogamer** [reply]
I'm curious to hear more about what you think Discord did wrong, particularly regarding threads - if you'd rather discuss this elsewhere that's fair

**Reply by nulldg** [reply]
thank you for indulging my rants haha :heart:
this is all just my opinion:
- threads in discord are presented badly. they're sprawled across the UI in such a redundant and inconsistent manner. despite essentially being mini-channels, threads are essentially invisible unless you spot it in the thread list (there is no hint when a new thread is made except in the chatlog) or in the chatlog (which gets buried into the backlog).
- viewing a thread you're not in randomly includes the thread in the channel list until you focus away, which adds to the inconsistency.
- you can add people to threads by mentioning them, and discord calls this "inviting" like they have any choice, but it actually forces them to autojoin even if they don't want to.
- longer auto-expiry times used to be pay-walled behind server boosts (discord backpedaled on that decision).
- they cannot be configured to autolock some time after closing, so users sometimes necrobump old threads, and it makes spam even more terrible when they go through the thread list and just crosspost through all the closed threads of every channel, causing them all to reopen.
- joining closed threads is buggy and offers little feedback, or none if you don't have perms to post in it.
- threads can be a huge nuisance for appbots to deal with and require some extra code to properly support.
- you can't even do anything interesting with threads like move them to a different channel or move a range of messages from a channel into a thread, so the problem-solving potential of the medium is seriously unrealized.
- the addition of threads stole development time and attention from channels, which are vastly more important. for example, it should be possible to archive a channel (plenty of servers want to change things and clean up, but don't want to destroy historical chatlogs), but instead this feature is reserved for threads for some reason.
- etc, etc.

imo threads should be simplified, and either made to resemble self-archiving mini channels, or made to be something totally different and unique. the way the "join/leave" mechanism works is the source of many of these issues, and i think it could be removed while still allowing users to show/pin/hide threads on the channel list.

if the revolt team do decide to add threads (please consider adding forums instead!), i beg the revolt team to please not copy the way discord did it. their design should be taken more as a cautionary tale.

**Reply by Rexogamer** [reply]
This is really interesting and useful - we have been planning to add threads/forums (the latter with a somewhat more traditional implementation) and detailed feedback like this is really appreciated :3
